File: Network/NetworkAndroid/app/src/main/java/com/bpxamarin/networkandroid/MainActivity.java
package com.bpxamarin.networkandroid;

import android.app.Activity;
import android.os.Bundle;
import android.os.StrictMode;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.google.gson.Gson;

import java.io.BufferedInputStream;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

public class MainActivity extends Activity {

    private static final String url = "https://raw.githubusercontent.com/211260fg/BPXamarin/master/README.md";
    private static final String jsonData =
        "{ " +
            "'firstname' : 'Florian', " +
            "'lastname' : 'Goeteyn', " +
            "'email' : 'florian_goeteyn@outlook.com', " +
            "'balance' : 12.5 " +
        "}";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Allow network calls on the main thread
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().permitAll().build());

        Button btnGet = findViewById(R.id.btn_get);
        Button btnConvertJson = findViewById(R.id.btn_convertjson);

        btnGet.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getTextFromUrlJava(url);
            }
        });

        btnConvertJson.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                convertJson(jsonData);
            }
        });
    }

    private void getTextFromUrlJava(String url) {
        HttpURLConnection urlConnection = null;
        StringBuilder result = new StringBuilder();
        try {
            URL connectionUrl = new URL(url);
            urlConnection = (HttpURLConnection) connectionUrl.openConnection();

            int code = urlConnection.getResponseCode();

            if (code == 200) {
                InputStream stream = new BufferedInputStream(urlConnection.getInputStream());
                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(stream));
                String line;

                while ((line = bufferedReader.readLine()) != null)
                    result.append(line);
                stream.close();
            }
            Toast.makeText(this, result.toString(), Toast.LENGTH_SHORT).show();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (urlConnection != null)
                urlConnection.disconnect();
        }
    }

    private void convertJson(String json){
        User user = new Gson().fromJson(json, User.class);

        String text =
                "First name: "+user.getFirstName()+"\n"+
                "Last name"+user.getLastName()+"\n"+
                "Email: "+user.getEmail()+"\n"+
                "Balance: "+user.getBalance();

        Toast.makeText(this, text, Toast.LENGTH_SHORT).show();
    }
}

File: Network/NetworkXamarin/App.Droid/MainActivity.cs
using System.IO;
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using Java.IO;
using Java.Net;
using Newtonsoft.Json;

namespace App.Droid
{
    [Activity(MainLauncher = true)]
    public class MainActivity : Activity, View.IOnClickListener
    {
        private static string Url => "https://raw.githubusercontent.com/211260fg/BPXamarin/master/README.md";
        private static string JsonData => 
            @"
                {
                    'firstname' : 'Florian',
                    'lastname': 'Goeteyn',
                    'email' : 'florian_goeteyn@outlook.com',
                    'balance' : 12.5
                }
            ";

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);
            
            //Allow network calls on the main thread
            StrictMode.SetThreadPolicy(new StrictMode.ThreadPolicy.Builder().PermitAll().Build());

            var btnGetJava = FindViewById(Resource.Id.btn_get_java);
            var btnGetNet = FindViewById(Resource.Id.btn_get_net);
            var btnConvertJson = FindViewById(Resource.Id.btn_convertjson);

            btnGetJava.SetOnClickListener(this);
            btnGetNet.SetOnClickListener(this);
            btnConvertJson.SetOnClickListener(this);
        }

        private void GetTextFromUrlNet(string url)
        {
            string result;
            using (var client = new System.Net.WebClient())
            {
                result = client.DownloadString(url);
            }
            Toast.MakeText(this, result, ToastLength.Short).Show();
        }

        private void GetTextFromUrlJava(string url)
        {
            HttpURLConnection urlConnection = null;
            var result = "";
            try
            {
                var connectionUrl = new URL(url);
                urlConnection = (HttpURLConnection)connectionUrl.OpenConnection();

                var code = urlConnection.ResponseCode;

                if (code == HttpStatus.Ok)
                {
                    var stream = new BufferedStream(urlConnection.InputStream);
                    var bufferedReader = new BufferedReader(new InputStreamReader(stream));
                    string line;

                    while ((line = bufferedReader.ReadLine()) != null)
                        result += line;
                    stream.Close();
                }

                Toast.MakeText(this, result, ToastLength.Short).Show();
            }
            finally
            {
                urlConnection?.Disconnect();
            }
        }

        private void ConvertJson(string json)
        {
            var user = JsonConvert.DeserializeObject<User>(json);

            var text = $@"
                          First name: {user.FirstName}
                          Last name: {user.LastName}
                          Email: {user.Email}
                          Balance: {user.Balance}
                        ";

            Toast.MakeText(this, text, ToastLength.Short).Show();
        }

        public void OnClick(View v)
        {
            switch (v.Id)
            {
                case Resource.Id.btn_get_java:
                    GetTextFromUrlJava(Url);
                    break;
                case Resource.Id.btn_get_net:
                    GetTextFromUrlNet(Url);
                    break;
                case Resource.Id.btn_convertjson:
                    ConvertJson(JsonData);
                    break;
            }
        }
    }
}

